src/main/scala/cromwell/webservice/CallCachingParameters.scala

```diff
@@ -0,0 +1,113 @@
+package cromwell.webservice
+
+import cromwell.engine.WorkflowId
+import cromwell.engine.db.DataAccess._
+import cromwell.engine.db.{DataAccess, ExecutionDatabaseKey}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+import scalaz.Scalaz._
+import scalaz.{Failure, Success, ValidationNel}
+
+case class CallCachingParameters private(workflowId: WorkflowId, callKey: Option[ExecutionDatabaseKey], allow: Boolean)
+
+object CallCachingParameters {
+
+  private [webservice] def validateRecognizedKeys(queryParameters: QueryParameters): ValidationNel[String, Unit] = {
```

Comment:
I meant something like this, although I don't care about the name of it as long as it makes sense. I've seen this, "ErrorOr", etc.

    type EValidation[+A] = Validation[String, A]